addons/stock_account/models/account_move.py

```diff
@@ -191,12 +191,12 @@ def _stock_account_anglo_saxon_reconcile_valuation(self, product=False):
                 continue
 
             products = product or move.mapped('invoice_line_ids.product_id')
-            for product in products:
-                if product.valuation != 'real_time':
+            for p in products:
```

Comment:
revert that stupid renaming of `product` by `p` for lisibility plz.

Reply:
Btw, after reading your diff, I don't see any difference except the renaming...

Reply:
The main difference lies on line 193:
`products = product or move.mapped('invoice_line_ids.product_id')`
The  `product` variable here is coming from the parameter of the method, and we use the same `product` variable on line 194. hence on the next invoice, we expect `products` on line 194 to be `move.mapped('invoice_line_ids.product_id')` but it is set to last product on previous invoice instead.
Hope you see the difference.

Reply:
@hendrikw25 OK I understand now. Can you explain the shadowing in your commit message plz?
I don't really like the `p` variable but I don't have any better idea...

Reply:
@smetl Sorry, I am not sure what you mean by "shadowing" in my commit message.
If you mean "Fix issue where auto reconciliation does not happen when selecting multiple invoices from the list and click action "Post Entries", it literally means that. 
From all selected invoices, the reconciliation only happens on the first invoice, and nothing on the second invoice and the rest.

> I don't really like the p variable but I don't have any better idea...

Yes, me too, maybe replace it to `pp `or `prod` or `l_product`

Reply:
I mean explaining the `product` variable in the loop is shadowing the `product` parameter and then, is triggering this issue.
Go for `prod` :)

Reply:
@smetl Sorry, I cannot find how to edit my commit message from github web interface.
Could you help me to replace the commit message? 

I am not sure how to rephrase my commit message in better way, you might need to help me :)
"product variable in the invoice line product loop shadowing the product variable in method parameter and causing the the wrong products assignment on the next invoice loop"

Thank you